core/Archive/ArchiveInvalidator.php

```diff
@@ -460,18 +460,12 @@ public function reArchiveReport($idSites, string $plugin, string $report = null,
     {
         $date2 = Date::yesterday();
 
+        $earliestDateToRearchive = $this->getEarliestDateToRearchive();
         if (empty($startDate)) {
-            $lastNMonthsToInvalidate = Config::getInstance()->General['rearchive_reports_in_past_last_n_months'];
-            if (empty($lastNMonthsToInvalidate)) {
-                return;
-            }
-
-            $lastNMonthsToInvalidate = (int) substr($lastNMonthsToInvalidate, 4);
-            if (empty($lastNMonthsToInvalidate)) {
-                return;
-            }
-
-            $startDate = $date2->subMonth($lastNMonthsToInvalidate)->setDay(1);
+            $startDate = $earliestDateToRearchive;
+        } else if (!empty($earliestDateToRearchive)) {
+            // don't allow archiving further back than the rearchive_reports_in_past_last_n_months date allows
+            $startDate = $startDate->isEarlier($earliestDateToRearchive) ? $earliestDateToRearchive : $startDate;
```

Comment:
Does that mean there won't be any possibility to rearchive specific reports for dates older than `rearchive_reports_in_past_last_n_months`? Wondering if there might be any use cases where it might make sense to force that somehow. But guess would be enough to implement it if we have such a use case at some point

Reply:
I guess a user could always configure it differently above setting.

Reply:
For this a plugin could manually invalidate, since rearchiveRepors is really just a smart wrapper for `markArchivesAsInvalidated`.

Reply:
👍